From c7ebbe76d60e68dc5fa3af2dcfa322affdb9bc88 Mon Sep 17 00:00:00 2001
From: upterm-ports <upterm-ports@localhost>
Date: Sat, 17 Oct 2026 11:18:57 +0200
Subject: [PATCH 2/3] compat/disk.h: NO_STATVFS for libcs without statvfs()

ixemul ships sys/statvfs.h (newlib) but no statvfs(); git diagnose then fails
with an error instead of failing to link.
---
 compat/disk.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/compat/disk.h b/compat/disk.h
index 23bc1be..f19a210 100644
--- a/compat/disk.h
+++ b/compat/disk.h
@@ -33,6 +33,10 @@ static int get_disk_info(struct strbuf *out)
 	strbuf_addf(out, "Available space on '%s': ", buf.buf);
 	strbuf_humanise_bytes(out, avail2caller.QuadPart);
 	strbuf_addch(out, '\n');
+#elif defined(NO_STATVFS)
+	error(_("could not determine free disk size: not supported on AmigaOS"));
+	res = -1;
+	goto cleanup;
 #else
 	struct statvfs stat;
 
-- 
2.50.1 (Apple Git-155)

